### DISCUSSION TITLE
Why does not pyinstaller support python 3.12 ?

### DISCUSSION THREAD

**Discussion by yuji38kwmt** [pyinstaller/pyinstaller#7605] [PyInstaller] [answered]
PyInstaller supports Python versions 3.7 through 3.11 .

```
[options]
python_requires = >=3.7, <3.12
```

https://github.com/pyinstaller/pyinstaller/blob/3dadde82a4162d3b0ec59e713eebbae1430a14c7/setup.cfg#L60

Why does not pyinstaller support python 3.12 ?
I think that upper bound of `python_requires` is python 4.0 in many library .

Python is backwards compatible. So I think that pyinstaller works fine is in python 3.12.

**Comment by bwoodsend** [comment, marked as answer]
Python's bytecode (which PyInstaller has to read) is not backwards compatible. Nor is its binary ABI which PyInstaller's bootloaders depend on. And in the case of Python 3.12, the standard lib libraries `imp` and `distutils` are being removed so PyInstaller needs some fairly serious work to remove its dependency on those.

**Reply by yuji38kwmt** [reply]
Thanks!
